ComponentKit/LayoutComponents/CKStackUnpositionedLayout.mm

```diff
@@ -351,17 +351,23 @@ static void flexChildrenAlongStackDimension(std::vector<CKStackUnpositionedItem>
     }
     return;
   }
-  std::function<CGFloat(const CKStackUnpositionedItem &, BOOL)> flexAdjustment = flexAdjustmentInViolationDirection(items,
-                                                                                                                    style,
-                                                                                                                    violation,
-                                                                                                                    flexFactorSum);
+  std::function<CGFloat(const CKStackUnpositionedItem &)> flexAdjustment = flexAdjustmentInViolationDirection(items,
+                                                                                                              style,
+                                                                                                              violation,
+                                                                                                              flexFactorSum);
+
+  // Compute any remaining violation. Any remaining violation it will be allocated to the first flexible child.
+  const CGFloat remainingViolation = std::accumulate(items.begin(), items.end(), violation, [&](CGFloat x, const CKStackUnpositionedItem &item) {
```

Comment:
It's a bit unfortunate to do a completely separate pass just to compute the remaining violation. As an alternative I can compute the remaining violation in the loop below and the apply it to the first flexible child.

This felt a bit cleaner though and I don't really anticipate this being too expensive.